matplotcheck/tests/test_base_data.py

```diff
@@ -72,6 +83,143 @@ def test_assert_xydata_xlabel_fails(pt_bar_plt, pd_df):
     plt.close()
 
 
+def test_assert_xydata_xlabel_text():
+    "Tests the xlabels flag on xydata works to test labels with text data"
+    data = {
+        "months": ["Jan", "Feb", "Mar", "Apr", "May", "June", "July"],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    df = pd.DataFrame(data)
+
+    fig, ax = plt.subplots()
+    df.plot("months", "data", kind="bar", ax=ax)
+    axis = plt.gca()
+
+    pt = PlotTester(axis)
+    pt.assert_xydata(df, xcol="months", ycol="data", xlabels=True)
+
+    plt.close()
+
+
+def test_assert_xydata_xlabel_text_fails():
+    "Tests the xlabels flag on xydata fails when testing labels with wrong text data"
+    correct_data = {
+        "months": ["Jan", "Feb", "Mar", "Apr", "May", "June", "July"],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    plot_data = {
+        "months": ["Jan", "Feb", "Mar", "Apr", "May", "June", "Sept"],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    correct_df = pd.DataFrame(correct_data)
+    plot_df = pd.DataFrame(plot_data)
+
+    fig, ax = plt.subplots()
+    plot_df.plot("months", "data", kind="bar", ax=ax)
+    axis = plt.gca()
+
+    pt = PlotTester(axis)
+    with pytest.raises(AssertionError, match="Incorrect data values"):
+        pt.assert_xydata(correct_df, xcol="months", ycol="data", xlabels=True)
+
+    plt.close()
+
+
+def test_assert_xydata_xlabel_numeric():
+    "Tests the xlabels flag on xydata works with numeric expected x-labels."
+    correct_data = {
+        "months": [1, 2, 3, 4, 5, 6, 7],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    plot_data = {
+        "months": [1, 2, 3, 4, 5, 6, 7],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    correct_df = pd.DataFrame(correct_data)
+    plot_df = pd.DataFrame(plot_data)
+
+    fig, ax = plt.subplots()
+    plot_df.plot("months", "data", kind="bar", ax=ax)
+    axis = plt.gca()
+
+    pt = PlotTester(axis)
+    # import pdb; pdb.set_trace()
+    pt.assert_xydata(correct_df, xcol="months", ycol="data", xlabels=True)
+    plt.close()
+
+
+def test_assert_xydata_xlabel_numeric_fails():
+    "Tests the xlabels flag on xydata correctly fails with numeric expected x-labels."
+    correct_data = {
+        "months": [1, 2, 3, 4, 5, 6, 7],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    plot_data = {
+        "months": [1, 2, 3, 4, 5, 6, 99999],
+        "data": [0.635, 0.795, 1.655, 3.085, 2.64, 1.44, 1.02],
+    }
+    correct_df = pd.DataFrame(correct_data)
+    plot_df = pd.DataFrame(plot_data)
+
+    fig, ax = plt.subplots()
```

Comment:
ok so looking at this, i think you hsould create 2 fixtues.

1. returns the plottester object for the plot_data dataframe and 
2. one that returns the dataframe used to test that the data are correct and to produce the plot. 

the first fixture above would actually required #2 above the data.frame used to test the data here.